CsharpExtrasTest/OneBased/OneBasedArray2DTest.cs

```diff
@@ -217,6 +217,44 @@ public void Given_2DArrayOfBytes_When_SubArray_Then_ExpectedSubArrayReturned
                 startAtRow, startAtColumn, stopBeforeRow, stopBeforeColumn));
         }
 
+        [Test]
+        public void Given_2DArrayOfByes_When_FlattenRowMajor_Then_ExpectedOneDimArrayReturned()
+        {
+            //Assemble
+            byte[,] zeroBased = new byte[,]
+            {
+                {00, 01, 02},
+                {10, 11, 12},
+                {20, 21, 22}
+            };
+            OneBasedArray2DImpl<byte> oneBased = new OneBasedArray2DImpl<byte>(zeroBased);
+
+            //Act
+            IOneBasedArray<byte> flattened = oneBased.FlattenRowMajor();
+
+            //Assert
+            Assert.AreEqual(new byte[] { 00, 01, 02, 10, 11, 12, 20, 21, 22 }, flattened.ZeroBasedEquivalent);
+        }
+
+        [Test]
+        public void Given_2DArrayOfByes_When_FlattenColumnRoMajor_Then_ExpectedOneDimArrayReturned()
+        {
+            //Assemble
+            byte[,] zeroBased = new byte[,]
+            {
+                {00, 01, 02},
+                {10, 11, 12},
+                {20, 21, 22}
+            };
+            OneBasedArray2DImpl<byte> oneBased = new OneBasedArray2DImpl<byte>(zeroBased);
+
+            //Act
+            IOneBasedArray<byte> flattened = oneBased.FlattenColumnMajor();
+
+            //Assert
+            Assert.AreEqual(new byte[] { 00, 10, 20, 01, 11, 21, 02, 12, 22 }, flattened.ZeroBasedEquivalent);
+        }
+

```

Comment:
I would make these tests parameterized to make it easier to add more test-cases later on.